packages/extension/src/background/actionHandlers.ts

```diff
@@ -49,21 +50,61 @@ export const handleActionApproval = async (
     }
 
     case "ALPH_TRANSACTION": {
-      const { signature: signatureOpt, ...transaction } =
-        additionalData as ReviewTransactionResult & { signature?: string }
+      const transactions = additionalData as ReviewTransactionResult[]
       try {
-        const { signature } = await executeTransactionAction(
-          transaction,
-          signatureOpt,
-          background,
-          transaction.params.networkId,
-        )
+        if (transactions.length === 0) {
+          return {
+            type: "ALPH_TRANSACTION_FAILED",
+            data: { actionHash, error: "No transactions to execute" },
+          }
+        }
+
+        let results: TransactionResult[]
+        if (transactions.length === 1) {
+          const { signature: signatureOpt, ...transaction } =
+            transactions[0] as ReviewTransactionResult & { signature?: string }
+
+          const { signature } = await executeTransactionAction(
+            transaction,
+            signatureOpt,
+            background,
+            transaction.params.networkId,
+          )
+
+          transactionWatcher.refresh()
+
+          results = [
+            {
+              type: transaction.type,
+              result: {
+                ...transaction.result,
+                signature
+              }
+            }
+          ] as TransactionResult[]
+        } else {
+          const { signatures } = await executeTransactionsAction(
+            transactions,
+            background,
+            transactions[0].params.networkId,
```

Comment:
Let's verify that all network IDs are the same before using the first one. Checking ahead of time can help prevent potential bugs, even though we'll also confirm in the Approval phase.